### ISSUE TITLE
Bug: false positive for Borrow trait check

### ISSUE THREAD

**Issue by thinkingInWorldByNull** [intellij-rust/intellij-rust#9947]
## Environment

IntelliJ Rust plugin version: 0.4.174.4743-221
Rust toolchain: on nightly, stable, 1.65.0, pretty much every version I try
IDE name and version: CLion 2022.1.3, Build #CL-221.5921.27
Operating system: Mac

## Problem description
False positive error check for Borrow trait check.

## Steps to reproduce

```rust
use std::borrow::Borrow;

fn false_position_fot_borrow_check() {
    let s = "hello world!".to_owned();

//❗️❗️ this is right, but rust plugin report error: mismatched types [E0308] expected `&String`, found `&str` 
    let r1: &String = s.borrow();     
    let r2: &str = s.borrow();

    println!("r1: {:p}, r2: {:p}", r1, r2);
}
```
